sql/src/main/java/io/crate/analyze/EvaluatingNormalizer.java

```diff
@@ -89,9 +89,20 @@ public EvaluatingNormalizer(Functions functions,
         }
     }
 
-    private abstract class BaseVisitor extends SymbolVisitor<TransactionContext, Symbol> {
+    private static class Context {
+
+        @Nullable
+        private final TransactionContext transactionContext;
+        private boolean replaceOnly = false;
```

Comment:
maybe extend variable name to replaceFieldsOnly?
Because if a function is normalized it's replaced too if ReplaceMode is Mutate.